.github/workflows/test.yml

```diff
@@ -0,0 +1,29 @@
+name: test
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Setup Go environment
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version: "stable"
+      - name: Run Unit Tests
+        run: go test -v ./server/...
+      - name: Run E2E Tests
+        run: |
+          go build -o fifo ./server
+          ./fifo >logs 2>&1 &
+          sleep 1
+          trap 'kill $!; cat logs' EXIT
+          go test -v ./client/...
+          cat logs
```

Comment:
aren't we printing the logs twice now? (once in the trap, once in the last line?)